youtube-client/src/app/auth/guards/loginguard.guard.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import {
+  CanActivate,
+  Router,
+  UrlTree,
+} from '@angular/router';
+import { Observable, of } from 'rxjs';
+import LoginService from '../services/login.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export default class LoginguardGuard implements CanActivate {
+  constructor(
+    private auth: LoginService,
+    private router: Router,
+  ) {}
+
+  canActivate(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
```

Comment:
update `type` of function. Since you always return `Observable<boolean>` it doesn't make sence to have different types included as "expected" when it's never going to happen in this case :)